lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/authentication/CachingKerberosAuthentication.java

```diff
@@ -37,11 +37,26 @@ public CachingKerberosAuthentication(KerberosAuthentication kerberosAuthenticati
 
     public synchronized Subject getSubject()
     {
-        if (subject == null || nextRefreshTime < System.currentTimeMillis()) {
+        if (subject == null || ticketNeedsRefresh()) {
             subject = requireNonNull(kerberosAuthentication.getSubject(), "kerberosAuthentication.getSubject() is null");
             KerberosTicket tgtTicket = getTicketGrantingTicket(subject);
             nextRefreshTime = KerberosTicketUtils.getRefreshTime(tgtTicket);
         }
         return subject;
     }
+
+    public synchronized void reauthenticateIfSoonWillBeExpired()
```

Comment:
The same `Subject` instance that was used to build the kudu client needs to have the new kerberos ticket associated with it. So the `getSubject` method does not work since it will create a new subject instance.